tools/crm_resource_runtime.c

```diff
@@ -262,33 +262,33 @@ cli_resource_update_attribute(pe_resource_t *rsc, const char *requested_name,
     pcmk__output_t *out = rsc->cluster->priv;
     int rc = pcmk_rc_ok;
 
-    char *local_attr_id = NULL;
+    char *found_attr_id = NULL;
     char *local_attr_set = NULL;
 
     GList/*<pe_resource_t*>*/ *resources = NULL;
-    const pe_resource_t *top = pe__const_top_resource(rsc, false);
+    const char *top_id = pe__const_top_resource(rsc, false)->id;
```

Comment:
I feel like we should be checking that rsc is not NULL, but a lot of these functions would have problems if that were the case.  Hopefully the caller handles it.

Reply:
Yeah, I usually prefer to assume non-NULL only for static functions since we can easily double-check all callers in that case. But these are used in only one spot outside this file, so it's not horrible ...